jobserv/api/worker.py

```diff
@@ -39,11 +47,37 @@ def wrapper(*args, **kwargs):
         if not key:
             return jsendify("No Authorization header provided", 401)
         parts = key.split(" ")
-        if len(parts) != 2 or parts[0] != "Token":
+        if len(parts) != 2 or parts[0] not in ("Token", "Bearer"):
             return jsendify("Invalid Authorization header", 401)
-        worker = get_or_404(Worker.query.filter_by(name=kwargs["name"], deleted=False))
-        if not worker.validate_api_key(parts[1]):
-            return jsendify("Incorrect API key for host", 401)
+
+        if parts[0] == "Bearer":
+            try:
+                w = worker_from_jwt(parts[1])
+            except PyJWTError as e:
+                return jsendify(str(e), 401)
+
+            if w.name != kwargs["name"]:
```

Comment:
I am too unfamiliar with this codebase, so I just thought I'd ask. Are both the JWT token and the `kwargs['name']` user input? Then you can just always make sure they are the same even though it is a bogus name as we will create the worker then regardless. Just making sure not anyone can just create a score of workers at will.

Reply:
You have to have a valid token to get this far. So its a token we trust. This is a check to ensure that worker-a can't steal work for worker-b. Especially in the case where worker-a is one company and worker-b is another company.

Reply:
I understand. :+1: